src/metatrain/utils/data/target_info.py

```diff
@@ -212,13 +230,33 @@ def is_compatible_with(self, other: "TargetInfo") -> bool:
             other_block = other.layout[key]
             if not block.samples == other_block.samples:
                 return False
-            if not block.components == other_block.components:
+            if not _is_labelslist_equal(block.components, other_block.components):
```

Comment:
Is there no mts function for this?

Reply:
Well this is also what `_is_labelslist_equal` does, but also checking that the lengths are equal otherwise with just this you might get that two different length lists are equal.

Reply:
Please don't use private functions, this is what leads to needing strong version pinning. 

But we can define a new public one for `check_same_metadata` or whatever the semantics needs to be